### PR TITLE
 Add highlight to differences #2

### LINKED ISSUE #2: Add highlight to differences
Examples:
image

image

### PR DESCRIPTION
**Pull request by ABenassi87**
Changes:
  - Added diffs highlight to Side by Side and Line by Line comparisons.

Resolves #2

### CODE REVIEW

**Comment by paustint** [comment]
@ABenassi87 - seems like the "copy" command is not cross-platform compatible (meaning all the package commands fail) - Maybe you can use something like [copyfiles](https://www.npmjs.com/package/copyfiles) or a JS script of some sort (gulp / release-ite /etc..)

**Comment by ABenassi87** [comment]
@paustint thanks Austin. I added copyfiles as a dependency and I used it to copy the files.

File: projects/ngx-text-diff/src/lib/ngx-text-diff.component.ts [hunk 1]
```diff
@@ -14,13 +14,14 @@ export class NgxTextDiffComponent implements OnInit {
   @Input() right = '';
   @Input() loading = false;
   @Input() diffContentObservable$: Observable<DiffContent>;
```
Comment (paustint) [inline projects/ngx-text-diff/src/lib/ngx-text-diff.component.ts]: @ABenassi87 - I am not really sure the convention here, but I have never seen an input name that has a `$`.  While I know it is convention for variable names, it feels a bit strange having this on an `@Input()` for some reason.

Also, I don't think it is convention to have the word `Observable` in the variable name.

Action:
1. I think renaming it to something like `diffContent` would be more straight forward to work with.
2. I don't see where this input is included in the `README.md` as an option. - We should add documentation to all available inputs and explain when to use each one.



_I tried to find some references, but did not find anything too solid_:
https://angular.io/guide/rx-library
https://material.angular.io/components/table/api (one variable input appears to be an observable)
https://material.angular.io/components/chips/api

File: projects/ngx-text-diff/src/lib/ngx-text-diff.component.ts [hunk 2]
```diff
@@ -14,13 +14,14 @@ export class NgxTextDiffComponent implements OnInit {
...
+  @Input() showBtnToolbar = true;
   subscriptions: Subscription[] = [];
```
Comment (paustint) [inline projects/ngx-text-diff/src/lib/ngx-text-diff.component.ts]: I see we are using this subscriptions array - but we are never unsubscribing once the component is destroyed. I would recommend to add `OnDestroy()` and unsubscribe from all subscriptions.

File: projects/ngx-text-diff/src/lib/ngx-text-diff.component.ts [hunk 3]
```diff
@@ -14,13 +14,14 @@ export class NgxTextDiffComponent implements OnInit {
   @Input() right = '';
   @Input() loading = false;
   @Input() diffContentObservable$: Observable<DiffContent>;
+  @Input() showBtnToolbar = true;
```
Comment (paustint) [inline projects/ngx-text-diff/src/lib/ngx-text-diff.component.ts]: Add to `README.md`

File: projects/ngx-text-diff/src/lib/ngx-text-diff.component.ts [hunk 4]
```diff
@@ -14,13 +14,14 @@ export class NgxTextDiffComponent implements OnInit {
   @Input() right = '';
   @Input() loading = false;
```
Comment (paustint) [inline projects/ngx-text-diff/src/lib/ngx-text-diff.component.ts]: Add to `README.md`

File: projects/ngx-text-diff/src/lib/ngx-text-diff.component.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import { ChangeDetectorRef, Component, Input, OnInit } from '@angular/core';
-import { DiffContent, DiffTableBothRow, DiffTableFormat, DiffTableFormatOption, DiffTableRow } from './ngx-text-diff.model';
+import { DiffContent, DiffLineResult, DiffPart, DiffTableFormat, DiffTableFormatOption, DiffTableRowResult } from './ngx-text-diff.model';
```
Comment (paustint) [inline projects/ngx-text-diff/src/lib/ngx-text-diff.component.ts]: It appears `DiffLineResult` is unused. All unused imports should be removed

File: projects/ngx-text-diff/src/public_api.ts [hunk 6]
```diff
@@ -4,4 +4,8 @@
 
 export * from './lib/ngx-text-diff.service';
```
Comment (paustint) [inline projects/ngx-text-diff/src/public_api.ts]: I did not look at all the packaged components - but do we need to export every file as a public file, or just the ones we want the user to able to use?

File: projects/ngx-text-diff/src/public_api.ts [hunk 7]
```diff
@@ -4,4 +4,8 @@
...
 export * from './lib/ngx-text-diff.component';
+export * from './lib/loader-sppiner/loader-sppiner.component';
```
Comment (paustint) [inline projects/ngx-text-diff/src/public_api.ts]:
```suggestion
export * from './lib/loader-spinner/loader-spinner.component';
```

**Comment by ABenassi87** [comment]
@paustint Thank you very much for code review and the time that you spent on doing it.

**Pull request merged**